### PR TITLE
Added update verify and create functions

### PR DESCRIPTION
**Pull request by Sri-Harsha06**
An api for update with upsert true to either update data or add if data does not exist 
An api to create or save a blueprint in the data base 
An api to verify if a particular project with a project id belongs to a user for giving the acces to edit only to the user to whom the project belongs to 
Moved the api to view a particular project from protected to public

### CODE REVIEW

File: designer/router.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 var wda = require('./controller');
 
 module.exports = function(router) {
-    router.get('/blueprint/:project_id',wda.getBlueprint);
     router.get('/blueprints',wda.getBlueprints);
     router.delete('/delete/:project_id', wda.deleteBlueprint);
     router.get('/projects', wda.getProjectNames);
+    router.post('/update/:project_id',wda.updateBlueprint);
+    router.get('/user/:project_id',wda.verifyproject);
```
Comment (craxkumar) [inline designer/router.js:8]: Follow Camel 🐪 casing for method name 
verifyProject

File: designer/router.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+    router.post('/create',wda.createBlueprint);
```
Comment (craxkumar) [inline designer/router.js:9]: Router can be called as 
/blueprints/create

File: designer/controller.js [hunk 3]
```diff
@@ -4,21 +4,106 @@ const jdlConverter = require('../utility/jsonToJdl');
...
+ * @param {*} req 
+ * @param {*} res 
+ */
+exports.verifyproject = function(req,res) {
+  const userId = req.kauth.grant.access_token.content.sub;
+  blueprintDao.getByProjectId({project_id: req.params.project_id})
+  .then(result => {
+    if (Array.isArray(result) && result.length === 1) {
+        var uniqueResult = result[0];
+        if(userId == uniqueResult.user_id){
```
Comment (craxkumar) [inline designer/controller.js:69]: No need to send the string called Yes or No u can simple use the http status 
So when it return 200 the http_status will be OK when it returns 204 the status will be NO_CONTENT u can either use http code or status no need to send the string externally 
Give it a try

File: designer/controller.js [hunk 4]
```diff
@@ -4,21 +4,106 @@ const jdlConverter = require('../utility/jsonToJdl');
...
+        else{
+          return res.status(204).send("No");
+        }
+      }
+  })
+  .catch(error => {
+    console.error("Error retrieving blueprint:", error);
+    return res.status(500).send({ message: "Error retrieving blueprint" });
+  });
+  // return res.status(204).send("No");
```
Comment (craxkumar) [inline designer/controller.js:90]: Remove this commented line if not required, please

**Review by craxkumar — CHANGES_REQUESTED** [review]

File: designer/router.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 var wda = require('./controller');
 
 module.exports = function(router) {
-    router.get('/blueprint/:project_id',wda.getBlueprint);
     router.get('/blueprints',wda.getBlueprints);
     router.delete('/delete/:project_id', wda.deleteBlueprint);
     router.get('/projects', wda.getProjectNames);
+    router.post('/update/:project_id',wda.updateBlueprint);
```
Comment (craxkumar) [inline designer/router.js:7]: so that i missed to point out 
but please change this url to 
/blueprints/update

**Review by craxkumar — CHANGES_REQUESTED** [review]

**Review by craxkumar — CHANGES_REQUESTED** [review]
Update the routers as disccused!

**Review by craxkumar — APPROVED** [review]
Review OK!

**Review by rohithteja-cmi — CHANGES_REQUESTED** [review]
Please format the code added as part of this PR

**Review by rohithteja-cmi — APPROVED** [review]

**Pull request merged**